web_src/js/components/ViewFileTree.vue

```diff
@@ -12,27 +20,244 @@ const props = defineProps({
+  searchQuery.value = (e.target as HTMLInputElement).value;
+  selectedIndex.value = 0;
+};
+
+const handleKeyDown = (e: KeyboardEvent) => {
+  if (e.key === 'Escape' && searchQuery.value) {
+    e.preventDefault();
+    clearSearch();
+    return;
+  }
+
+  if (!searchQuery.value || filteredFiles.value.length === 0) return;
+
+  if (e.key === 'ArrowDown') {
+    e.preventDefault();
+    selectedIndex.value = Math.min(selectedIndex.value + 1, filteredFiles.value.length - 1);
+    scrollSelectedIntoView();
+  } else if (e.key === 'ArrowUp') {
+    e.preventDefault();
+    selectedIndex.value = Math.max(selectedIndex.value - 1, 0);
+    scrollSelectedIntoView();
+  } else if (e.key === 'Enter') {
+    e.preventDefault();
+    const selectedFile = filteredFiles.value[selectedIndex.value];
+    if (selectedFile) {
+      handleSearchResultClick(selectedFile.matchResult.join(''));
+    }
+  }
+};
+
+const clearSearch = () => {
+  searchQuery.value = '';
+  if (searchInputElement) searchInputElement.value = '';
+};
+
+const scrollSelectedIntoView = () => {
+  nextTick(() => {
+    const resultsEl = searchResults.value;
+    if (!resultsEl) return;
+    
+    const selectedEl = resultsEl.querySelector('.file-tree-search-result-item.selected');
+    if (selectedEl) {
+      selectedEl.scrollIntoView({block: 'nearest', behavior: 'smooth'});
+    }
+  });
+};
+
+const handleClickOutside = (e: MouseEvent) => {
+  if (!searchQuery.value) return;
+  
+  const target = e.target as HTMLElement;
+  const resultsEl = searchResults.value;
+  
+  if (searchInputElement && !searchInputElement.contains(target) && 
+      resultsEl && !resultsEl.contains(target)) {
+    clearSearch();
+  }
+};
+
 onMounted(async () => {
   store.rootFiles = await store.loadChildren('', props.treePath);
   elRoot.value.closest('.is-loading')?.classList?.remove('is-loading');
+  
+  // Load all files for search
+  const treeListUrl = elRoot.value.closest('#view-file-tree')?.getAttribute('data-tree-list-url');
+  if (treeListUrl) {
+    const response = await GET(treeListUrl);
+    allFiles.value = await response.json();
+  }
+  
+  searchInputElement = document.querySelector('#file-tree-search');
+  if (searchInputElement) {
+    searchInputElement.addEventListener('input', handleSearchInput);
+    searchInputElement.addEventListener('keydown', handleKeyDown);
+  }
+  
+  document.addEventListener('click', handleClickOutside);
+  
   window.addEventListener('popstate', (e) => {
     store.selectedItem = e.state?.treePath || '';
     if (e.state?.url) store.loadViewContent(e.state.url);
   });
 });
+
+// Position search results below the input
+watch(searchQuery, async () => {
+  if (searchQuery.value && searchInputElement) {
+    await nextTick();
+    const resultsEl = searchResults.value;
+    if (resultsEl) {
+      const rect = searchInputElement.getBoundingClientRect();
+      resultsEl.style.top = `${rect.bottom + 4}px`;
+      resultsEl.style.left = `${rect.left}px`;
+    }
```

Comment:
Can this be solved with static `tw-` classes? Inline styles are problematic because they require CSP `style-src: unsafe-inline`, so we want to get rid of all of them.

https://v3.tailwindcss.com/docs/top-right-bottom-left

Reply:
> because they require CSP `style-src: unsafe-inline`

I think we have discussed before, `el.style` doesn't conflict with any CSP?